src/Translator.php

```diff
@@ -0,0 +1,542 @@
+            $this->generateJavascript();
+        }
+        if ($this->doTransifexPush) {
+            $this->transifexPushSource();
+        }
+        $this->gitCommitPushAndPullRequest();
+        $this->outputPullRequestUrls();
+    }
+
+    private function checkEnv(): void
+    {
+        $txInstructions = 'Install the new go version of the client https://developers.transifex.com/docs/cli';
+        if ($this->exec('which tx') === '') {
+            throw new Exception("Could not find tx executable. $txInstructions");
+        }
+        $help = $this->exec('tx help');
+        $help = str_replace(["\n", ' '], '', $help);
+        preg_match('#VERSION:([0-9]+\.[0-9]+)#', $help, $matches);
+        if (($matches[1] ?? 0) < 1.6) {
+            throw new Exception("Your version of tx is too old. $txInstructions");
+        }
+        if ($this->exec('which wget') === '') {
+            throw new Exception('Could not find wget command. Please install it.');
+        }
+        $this->txSite = getenv('TX_SITE');
+        if (!$this->txSite) {
+            throw new Exception('TX_SITE environment variable is not defined');
+        }
+        if (strpos($this->txSite, 'http://') !== 0 && strpos($this->txSite, 'https://') !== 0) {
+            $this->txSite = 'http://' . $this->txSite;
+        }
+        if (!filter_var($this->txSite, FILTER_VALIDATE_URL)) {
+            throw new Exception('SITE environment variable is not a valid url');
+        }
+        $this->githubToken = getenv('GITHUB_API_TOKEN');
+        if (!$this->githubToken) {
+            $message = implode(' ', [
+                'Could not get a valid token from GITHUB_API_TOKEN environment variable.',
+                'Create a new token with the public_repo checkbox ticked that will allow you to create pull-requests.'
+            ]);
+            throw new Exception($message);
+        }
+        // This will be set to true by default if TX_PULL environment variable is not defined
+        $this->doTransifexPullAndUpdate = !in_array(strtolower(getenv('TX_PULL')), ['false', '0']);
+        // This will be set to false by default if TX_PUSH environment variable is not defined
+        $this->doTransifexPush = in_array(strtolower(getenv('TX_PUSH')), ['true', '1']);
+        // This will be set to false by default if TX_DEV_MODE environment variable is not defined
+        $this->isDevMode = in_array(strtolower(getenv('TX_DEV_MODE')), ['true', '1']);
+        $this->log('');
+        $txt = $this->isDevMode ? 'ON (changes will not be pushed)' : 'OFF (changes will be pushed!)';
+        $this->log("TX_DEV_MODE is $txt");
+        $this->log('');
+    }
+
+    private function setModulePaths(): void
+    {
+        $vendorAllowList = [
+            'silverstripe',
+            'cwp',
+            'symbiote',
+            'dnadesign'
+        ];
+        $vendorDir = dirname(dirname(dirname(__DIR__)));
+        foreach (scandir($vendorDir) as $vendor) {
```

Comment:
Explicitly _don't_ scan over `.` and `..`
```suggestion
        foreach (array_diff(scandir($vendorDir), ['..',  '.']) as $vendor) {
```
The `in_array` check implicitly skips these anyway, but this is robust against that logic being removed or changing in the future.

Reply:
We don't need these there's an in_array() statement below it that does the same thing

Reply:
> The `in_array` check implicitly skips these anyway, but this is robust against that logic being removed or changing in the future.

My above comment still stands. Better to be explicit so if anything changes with the following logic we don't accidentally start looping over dots.

Reply:
Have refactored to use a private method to do this